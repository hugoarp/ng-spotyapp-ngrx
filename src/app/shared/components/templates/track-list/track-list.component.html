<section class="track-list">
  <h2 class="track-list__title">{{ listTitle }}</h2>
  <ng-content #default></ng-content>
  <article class="track-list__content">
    <app-track-details
      *ngFor="let track of tracks"
      [track]="track"
      (favoriteEvent)="emitFavoriteEvent($event)"
    ></app-track-details>
  </article>
</section>
