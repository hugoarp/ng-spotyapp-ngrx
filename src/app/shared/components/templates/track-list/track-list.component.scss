@import "variables";
@import "mixins";

.track-list {
  padding: 1rem;
  margin-bottom: 5rem;
  @include from($medium) {
    margin-bottom: 0;
    margin-left: 12rem;
  }
  &__title {
    margin-bottom: 1rem;
  }
  &__content {
    display: grid;
    justify-items: center;
    gap: 1rem;
    @include from($small) {
      grid-template-columns: repeat(2, 1fr);
    }
    @include from($medium) {
      grid-template-columns: repeat(3, 1fr);
    }
    @include from($x-large) {
      grid-template-columns: repeat(4, 1fr);
    }
    @include from($xxx-large) {
      grid-template-columns: repeat(5, 1fr);
    }
    @include from($high-quality) {
      grid-template-columns: repeat(6, 1fr);
    }
  }
}
