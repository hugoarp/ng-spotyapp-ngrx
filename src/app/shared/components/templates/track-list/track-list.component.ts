import { Component, OnInit, Input } from '@angular/core';
import { TrackInfo } from 'src/app/core/services/interfaces/track';

@Component({
  selector: 'app-track-list',
  templateUrl: './track-list.component.html',
  styleUrls: ['./track-list.component.scss'],
})
export class TrackListComponent implements OnInit {
  @Input() tracks: TrackInfo[] = [];
  @Input() listTitle: string = '';

  constructor() {}

  ngOnInit(): void {}

  emitFavoriteEvent(event: any) {
    console.log(event);
  }
}
