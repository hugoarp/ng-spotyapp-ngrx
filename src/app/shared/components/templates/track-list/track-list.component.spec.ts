import { ComponentFixture, TestBed } from '@angular/core/testing';

import { TrackListComponent } from './track-list.component';

describe('TrackListComponent', () => {
  let component: TrackListComponent;
  let fixture: ComponentFixture<TrackListComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [TrackListComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(TrackListComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe hacer match con el snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  test('debe emitir el evento favoriteEvent', () => {
    const track = {
      cover: 'cover',
      title: 'title',
      description: 'description',
      id: '1',
      favorite: false,
    };

    component.favoriteEvent.subscribe((trackInfo) => {
      expect(trackInfo).toEqual(track);
    });

    component.emitFavoriteEvent(track);
  });
});
