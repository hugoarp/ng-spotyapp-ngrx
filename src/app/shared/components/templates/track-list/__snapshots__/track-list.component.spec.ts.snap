// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackListComponent debe hacer match con el snapshot 1`] = `
<div
  id="root1"
>
  <section
    class="track-list"
  >
    <h2
      class="track-list__title"
    >
      
    </h2>
    <article
      class="track-list__content"
    >
      
    </article>
  </section>
</div>
`;
